src/main/java/com/manywho/services/sharepoint/client/HttpClient.java

```diff
@@ -26,11 +26,27 @@ public String executeRequest(HttpRequestBase requestBase){
 
                 if (status >= 200 && status < 300) {
                     HttpEntity entity = httpResponse.getEntity();
-
                     return EntityUtils.toString(entity);
                 }
 
-                throw new RuntimeException(httpResponse.getStatusLine().toString());
+                String errorLine = httpResponse.getStatusLine().toString();
+
+                // If we get an error about the admin consent, we send a better error description
+                if (httpResponse.getEntity() != null) {
+                    try {
+                        String entityError = EntityUtils.toString(httpResponse.getEntity());
+                        JSONObject objectError = new JSONObject(entityError);
+                        String errorDescription = objectError.getString("error_description");
+
+                        if (errorDescription.contains("AADSTS65001")) {
+                            errorLine = "The user or administrator has not consented to use this application." +
+                                    " Send an interactive authorization request for this user and resource. ";
+                        }
+                    } catch (Exception ignored){
```

Comment:
I don't think we should be ignoring this exception. We should at least be logging it, and probably displaying a fallback to the user if it's something we don't want them to see

Reply:
@jonjomckay yes it make sense, it says int the docs that I shouldn't send the errors to the user, so I am not doing it but I have added also the credentials error as I expect they would be the most common errors.

Reply:
Hmm, I'm still not sure that we should be ignoring the exception. What does @jacek-wojciechowski-dell think?

Reply:
@jonjomckay @josecollazzi 
So... if understand this correctly we mute exceptions that are being thrown while trying to get additional error details to send to the user. So catching this exception there is definitely correct - as we don't want them to spoil the main exception being thrown. And `getString` might throw an exception. So I think we should differentiate between

1. the expected data "error" (like "error_description" field might just not be there and IF it's OK, then we just mute it - as a way to deal with the JSON lib API that throws exceptions instead of returning nulls; I think we shouldn't log it as it's the equivalent of somebody forgetting to provide a mandatory value in the API - such logs are not useful)
2. unexpected error - and we should log it, so we know we are doing something wrong; but I don't know if in this code there is any unexpected error to deal with

So that's my point: differentiate between expected and log only unexpected ;)

Reply:
@jacek-wojciechowski-dell, @jonjomckay  I have removed the logs lines for common errors